platform/darwin/src/MGLLogging.m

```diff
@@ -0,0 +1,123 @@
+#import "MGLLogging_Private.h"
+
+#import <os/log.h>
+
+static os_log_t mapbox_log;
+
+@implementation MGLLogging
+
++ (void)initialize
+{
+    mapbox_log = os_log_create("com.mapbox.maps-ios-sdk", "SDK");
+}
+
++ (instancetype)sharedInstance
+{
+    static dispatch_once_t once;
+    static id sharedInstance;
+    dispatch_once(&once, ^{
+        sharedInstance = [[self alloc] init];
```

Comment:
Using a protocol breaks my intention for lightweight. Instead of using the block developers would have to implement a class with a delegate and set this object as the logger's delegate.